miniapps/tools/least-squares.cpp

```diff
@@ -0,0 +1,1720 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+
+namespace mfem::reconstruction
+{
+
+/// Enumerator for different function examples
+enum ExamplesType
+{
+   plane,
+   sine,
+   exp_sine,
+   num_examples  // last
+};
```

Comment:
I would consider partitioning this file in general into a header file with these enum, class, and functions definitions.